src/internal/kopia/base_finder.go

```diff
@@ -304,30 +307,108 @@ func (b *baseFinder) findBasesInSet(
 		// If we've made it to this point then we're considering the backup
 		// complete as it has both an item data snapshot and a backup details
 		// snapshot.
+		// Check first if this is an assist backup. Given we may have multiple
+		// assist backups, here are the rules for selecting one:
+		// 1. We must select at most one assist backup per reason tuple.
+		// 2. It must be the most recent assist backup for the reason.
+		// 3. It must be more recent than the merge backup if a merge backup
+		// exists.
+		// If an assist backup is found, the details snapshot associated with
+		// it will be utilized later during details merge process.
+		if assistBackup == nil {
+			assistBackup = b.checkForAssistBackup(ictx, bup, reason)
+			if assistBackup != nil {
+				logger.Ctx(ictx).Infow(
+					"found assist backup",
+					"assist_backup_id", bup.ID)
+
+				continue
+			}
+		}
+
+		// Found the most recent merge backup. We can stop here.
+		// TODO(pandeyabs): Ideally we would do some sanity checks here, e.g.
+		// 1. Check for MergeBackup tag.
+		// 2. Check for ErrorCount == 0.
+		// 1 cannot be done yet since we may have merge backups with older version
+		// of corso which won't have the tag.
+		// 2 cannot be verified yet because we support BestEffort mode which
+		// persists backup even if recoverable errors are encountered.
 		logger.Ctx(ictx).Infow("found complete backup", "base_backup_id", bup.ID)
 
 		me := ManifestEntry{
 			Manifest: man,
 			Reasons:  []Reasoner{reason},
 		}
+
 		kopiaAssistSnaps = append(kopiaAssistSnaps, me)
 
 		return &BackupEntry{
-			Backup:  bup,
-			Reasons: []Reasoner{reason},
-		}, &me, kopiaAssistSnaps, nil
+				Backup:  bup,
+				Reasons: []Reasoner{reason},
+			},
+			&me,
+			kopiaAssistSnaps,
+			assistBackup,
+			nil
 	}
 
 	logger.Ctx(ctx).Info("no base backups for reason")
 
-	return nil, nil, kopiaAssistSnaps, nil
+	return nil, nil, kopiaAssistSnaps, nil, nil
+}
+
+// checkForAssistBackup checks if the provided backup is an assist backup.
+func (b *baseFinder) checkForAssistBackup(
+	ctx context.Context,
+	bup *backup.Backup,
+	r Reasoner,
+) *BackupEntry {
+	if bup == nil {
+		return nil
+	}
+
+	allTags := map[string]string{
+		model.BackupTypeTag: model.AssistBackup,
+	}
+
+	allTags = normalizeTagKVs(allTags)
```

Comment:
I don't think manifest tags are normalized like snapshot tags are. The code for them was written at different times and I was trying to be fancy and get kopia CLI filters working with snapshot tags 😓

Reply:
Thanks! I'll remove this.